app/src/main/java/com/android/solvit/shared/model/NotificationsRepositoryFirestore.kt

```diff
@@ -92,13 +95,41 @@ class NotificationsRepositoryFirestore(private val db: FirebaseFirestore) :
       val message = document.getString("message") ?: return null
       val timestamp = document.getTimestamp("timestamp") ?: return null
       val isRead = document.getBoolean("isRead") ?: false
+      // Get ServiceRequest fields
+      val serviceRequestMap = document.get("serviceRequest") as? Map<String, Any> ?: return null
+      val locationMap = serviceRequestMap["location"] as? Map<String, Any>
+      val location =
+          if (locationMap != null) {
+            Location(
+                latitude = locationMap["latitude"] as? Double ?: 0.0,
+                longitude = locationMap["longitude"] as? Double ?: 0.0,
+                name = locationMap["name"] as? String ?: "")
+          } else {
+            null
+          }
+      val serviceRequest =
+          ServiceRequest(
+              uid = serviceRequestMap["uid"] as String,
+              title = serviceRequestMap["title"] as String,
+              type = Services.valueOf(serviceRequestMap["type"] as String),
+              description = serviceRequestMap["description"] as String,
+              userId = serviceRequestMap["userId"] as String,
+              providerId = serviceRequestMap["providerId"] as? String,
+              dueDate = serviceRequestMap["dueDate"] as Timestamp,
+              meetingDate = serviceRequestMap["meetingDate"] as? Timestamp,
+              location = location,
+              imageUrl = serviceRequestMap["imageUrl"] as? String,
+              packageId = serviceRequestMap["packageId"] as? String,
+              agreedPrice = serviceRequestMap["agreedPrice"] as? Double,
+              status = ServiceRequestStatus.valueOf(serviceRequestMap["status"] as String))
 
       Notification(
           uid = id,
           providerId = providerId,
           title = title,
           message = message,
           timestamp = timestamp,
+          serviceRequest = serviceRequest,
           isRead = isRead)
```

Comment:
It is probably more clean to store only the request uid and use the request repository to retrieve it, did you consider this?